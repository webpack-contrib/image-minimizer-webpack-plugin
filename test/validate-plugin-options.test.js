import ImageMinimizerPlugin from "../src";

describe("validate plugin options", () => {
  it("should work", () => {
    /* eslint-disable no-new */
    expect(() => {
      new ImageMinimizerPlugin({ test: /foo/ });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: "foo" });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ test: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ test: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ include: /foo/ });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: "foo" });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ include: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ include: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: /foo/ });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: "foo" });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: [/foo/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: [/foo/, /bar/] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: ["foo", "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: [/foo/, "bar"] });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ exclude: [true] });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ filter: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ filter: {} });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ severityError: false });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ severityError: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ severityError: "error" });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ severityError: () => {} });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: {} });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: { filter: () => true } });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: { filter: () => false } });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: null });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({
        minimizerOptions: { plugins: [] },
      });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({
        minimizerOptions: { filename: "[name].[ext]" },
      });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({
        minimizerOptions: { filename: () => "[name].[ext]" },
      });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: { filename: true } });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ minimizerOptions: { filename: {} } });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({
        minimizerOptions: { deleteOriginalAssets: false },
      });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({
        minimizerOptions: { deleteOriginalAssets: {} },
      });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ maxConcurrency: 2 });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ maxConcurrency: true });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ maxConcurrency: "true" });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({ loader: false });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ loader: true });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ loader: "true" });
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new ImageMinimizerPlugin({
        minify: [ImageMinimizerPlugin.imageminMinify],
      });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minify: ImageMinimizerPlugin.imageminMinify });
    }).not.toThrow();

    expect(() => {
      new ImageMinimizerPlugin({ minify: "true" });
    }).toThrowErrorMatchingSnapshot();
    /* eslint-enable no-new */
  });
});
