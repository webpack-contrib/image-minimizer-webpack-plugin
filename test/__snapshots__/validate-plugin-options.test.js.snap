// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate plugin options should work 1`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#test
   Details:
    * options.test should be an array:
      [RegExp | non-empty string, ...]
    * options.test should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test should be an instance of RegExp.
       * options.test should be a non-empty string."
`;

exports[`validate plugin options should work 2`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#test
   Details:
    * options.test[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test[0] should be an instance of RegExp.
       * options.test[0] should be a non-empty string."
`;

exports[`validate plugin options should work 3`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#include
   Details:
    * options.include should be an array:
      [RegExp | non-empty string, ...]
    * options.include should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include should be an instance of RegExp.
       * options.include should be a non-empty string."
`;

exports[`validate plugin options should work 4`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#include
   Details:
    * options.include[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include[0] should be an instance of RegExp.
       * options.include[0] should be a non-empty string."
`;

exports[`validate plugin options should work 5`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#exclude
   Details:
    * options.exclude should be an array:
      [RegExp | non-empty string, ...]
    * options.exclude should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude should be an instance of RegExp.
       * options.exclude should be a non-empty string."
`;

exports[`validate plugin options should work 6`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#exclude
   Details:
    * options.exclude[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude[0] should be an instance of RegExp.
       * options.exclude[0] should be a non-empty string."
`;

exports[`validate plugin options should work 7`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.severityError should be one of these:
   \\"off\\" | \\"warning\\" | \\"error\\"
   -> Allows to choose how errors are displayed.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#severityerror"
`;

exports[`validate plugin options should work 8`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.severityError should be one of these:
   \\"off\\" | \\"warning\\" | \\"error\\"
   -> Allows to choose how errors are displayed.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#severityerror"
`;

exports[`validate plugin options should work 9`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.severityError should be one of these:
   \\"off\\" | \\"warning\\" | \\"error\\"
   -> Allows to choose how errors are displayed.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#severityerror"
`;

exports[`validate plugin options should work 10`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.deleteOriginalAssets should be a boolean.
   -> Allows to remove original assets after minimization.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#deleteoriginalassets"
`;

exports[`validate plugin options should work 11`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizer should be one of these:
   [object { implementation?, filter?, filename?, options? }, ...] (should not have fewer than 1 item) | object { implementation?, filter?, filename?, options? }
   -> Allows you to setup the minimizer function and options.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#minimizer
   Details:
    * options.minimizer should be an array:
      [object { implementation?, filter?, filename?, options? }, ...] (should not have fewer than 1 item)
    * options.minimizer should be an object:
      object { implementation?, filter?, filename?, options? }"
`;

exports[`validate plugin options should work 12`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizer.filter should be an instance of function.
   -> Allows filtering of images."
`;

exports[`validate plugin options should work 13`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizer[0].filter should be an instance of function.
   -> Allows filtering of images."
`;

exports[`validate plugin options should work 14`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizer should be one of these:
   [object { implementation?, filter?, filename?, options? }, ...] (should not have fewer than 1 item) | object { implementation?, filter?, filename?, options? }
   -> Allows you to setup the minimizer function and options.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#minimizer
   Details:
    * options.minimizer.filename should be one of these:
      non-empty string | function
      -> Allows to set the filename for the minimized asset.
      Details:
       * options.minimizer.filename should be a non-empty string.
       * options.minimizer.filename should be an instance of function."
`;

exports[`validate plugin options should work 15`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizer should be one of these:
   [object { implementation?, filter?, filename?, options? }, ...] (should not have fewer than 1 item) | object { implementation?, filter?, filename?, options? }
   -> Allows you to setup the minimizer function and options.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#minimizer
   Details:
    * options.minimizer.filename should be one of these:
      non-empty string | function
      -> Allows to set the filename for the minimized asset.
      Details:
       * options.minimizer.filename should be a non-empty string.
       * options.minimizer.filename should be an instance of function."
`;

exports[`validate plugin options should work 16`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.generator should be an array:
   [object { implementation, type?, preset?, options?, filter?, filename? }, ...] (should not have fewer than 1 item)
   -> Allows you to setup the generator function and options.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#generator"
`;

exports[`validate plugin options should work 17`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.generator[0].filename should be one of these:
   non-empty string | function
   -> Allows to set the filename for the minimized asset.
   Details:
    * options.generator[0].filename should be a non-empty string.
    * options.generator[0].filename should be an instance of function."
`;

exports[`validate plugin options should work 18`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.generator[0].filter should be an instance of function.
   -> Allows filtering of images."
`;

exports[`validate plugin options should work 19`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.generator[0].type should be one of these:
   \\"import\\" | \\"asset\\"
   -> Type of generation
 - options.generator[0].filename should be one of these:
   non-empty string | function
   -> Allows to set the filename for the minimized asset.
   Details:
    * options.generator[0].filename should be a non-empty string.
    * options.generator[0].filename should be an instance of function."
`;

exports[`validate plugin options should work 20`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.concurrency should be a number.
   -> Number of concurrency optimization processes in one time.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#concurrency"
`;

exports[`validate plugin options should work 21`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.concurrency should be a number.
   -> Number of concurrency optimization processes in one time.
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#concurrency"
`;

exports[`validate plugin options should work 22`] = `
"Invalid options object. Image Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.loader should be a boolean.
   -> Automatically adding \`imagemin-loader\` (require for minification images using in \`url-loader\`, \`svg-url-loader\` or other).
   -> Read more at https://github.com/webpack-contrib/image-minimizer-webpack-plugin#loader"
`;
